src/execution/index/art/node16.cpp

```diff
@@ -42,67 +52,144 @@ idx_t Node16::GetNextPos(idx_t pos) {
 	return pos < count ? pos : DConstants::INVALID_INDEX;
 }
 
-unique_ptr<Node> *Node16::GetChild(idx_t pos) {
+Node *Node16::GetChild(ART &art, idx_t pos) {
 	D_ASSERT(pos < count);
-	return &child[pos];
+	Node::UnswizzleChild(art, children[pos]);
+	return (Node *)children[pos].pointer;
 }
 
 idx_t Node16::GetMin() {
 	return 0;
 }
 
-void Node16::Insert(ART &art, unique_ptr<Node> &node, uint8_t key_byte, unique_ptr<Node> &child) {
-	Node16 *n = static_cast<Node16 *>(node.get());
+void Node16::ReplaceChildPointer(idx_t pos, Node *node) {
+	AssignPointer(children[pos], node);
+}
+
+void Node16::Insert(Node *&node, uint8_t key_byte, Node *child) {
+	Node16 *n = (Node16 *)node;
 
 	if (n->count < 16) {
 		// Insert element
 		idx_t pos = 0;
 		while (pos < node->count && n->key[pos] < key_byte) {
 			pos++;
 		}
-		if (n->child[pos] != nullptr) {
+		if (n->children[pos] != 0) {
 			for (idx_t i = n->count; i > pos; i--) {
 				n->key[i] = n->key[i - 1];
-				n->child[i] = move(n->child[i - 1]);
+				n->children[i] = n->children[i - 1];
 			}
 		}
 		n->key[pos] = key_byte;
-		n->child[pos] = move(child);
+		AssignPointer(n->children[pos], child);
 		n->count++;
 	} else {
 		// Grow to Node48
-		auto new_node = make_unique<Node48>(art, n->prefix_length);
+		auto new_node = new Node48(n->prefix_length);
 		for (idx_t i = 0; i < node->count; i++) {
 			new_node->child_index[n->key[i]] = i;
-			new_node->child[i] = move(n->child[i]);
+			new_node->children[i] = n->children[i];
+			n->children[i] = 0;
 		}
-		CopyPrefix(art, n, new_node.get());
+		CopyPrefix(n, new_node);
 		new_node->count = node->count;
-		node = move(new_node);
+		delete node;
+		node = new_node;
 
-		Node48::Insert(art, node, key_byte, child);
+		Node48::Insert(node, key_byte, child);
 	}
 }
 
-void Node16::Erase(ART &art, unique_ptr<Node> &node, int pos) {
-	Node16 *n = static_cast<Node16 *>(node.get());
+DiskPosition Node16::Serialize(ART &art, duckdb::MetaBlockWriter &writer) {
```

Comment:
The `Serialize` and `Deserialize` functions look pretty much the same for all the node types, would it maybe make sense to template them? This seems to be a lot of duplicate code, but maybe templating it would be an unnecessary overhead...